File: packages/youtube-player/ios/Plugin/Classes/Results/CreatePlayerResult.swift
import Capacitor
import Foundation

@objc public class CreatePlayerResult: NSObject, Result {
    let id: String

    init(id: String) {
        self.id = id
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["id"] = id
        return result as AnyObject
    }
}

File: packages/youtube-player/android/src/main/java/io/capawesome/capacitorjs/plugins/youtubeplayer/classes/results/CreatePlayerResult.java
package io.capawesome.capacitorjs.plugins.youtubeplayer.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.youtubeplayer.interfaces.Result;

public class CreatePlayerResult implements Result {

    @NonNull
    private final String id;

    public CreatePlayerResult(@NonNull String id) {
        this.id = id;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("id", id);
        return result;
    }
}
